packages/app/src/cli/services/environment/id-matching.ts

```diff
@@ -1,45 +1,55 @@
 import {ExtensionRegistration} from '../dev/create-extension.js'
 import {IdentifiersExtensions} from '../../models/app/identifiers.js'
-import {Extension} from '../../models/app/extensions.js'
 import {err, ok, Result} from '@shopify/cli-kit/common/result'
+import {error, string} from '@shopify/cli-kit'
+import {ExtensionTypes} from '../../constants.js'
+import {MatchingError, RemoteRegistration} from './identifiers.js'
 
 export interface MatchResult {
   identifiers: IdentifiersExtensions
-  pendingConfirmation: {extension: Extension; registration: ExtensionRegistration}[]
-  toCreate: Extension[]
-  toManualMatch: {local: Extension[]; remote: ExtensionRegistration[]}
+  pendingConfirmation: {extension: LocalExtension; registration: ExtensionRegistration}[]
+  toCreate: LocalExtension[]
+  toManualMatch: {local: LocalExtension[]; remote: ExtensionRegistration[]}
+}
+
+export interface LocalExtension {
+  localIdentifier: string
+  graphQLType: string
+  type: ExtensionTypes
+  configuration: {name: string}
 }
 
 export async function automaticMatchmaking(
-  localExtensions: Extension[],
+  localExtensions: LocalExtension[],
   remoteRegistrations: ExtensionRegistration[],
   identifiers: {[localIdentifier: string]: string},
-): Promise<Result<MatchResult, Error>> {
-  const invalidEnvironmentError = err(new Error('invalid-environment'))
-
+  registrationIdField: 'id' | 'uuid',
```

Comment:
Not feeling strongly about it, I added the comment because I saw this used many times in tests, but there it's probably better to be explicit. Let's keep it without a default 👍